ENIGMAsystem/SHELL/Universal_System/Extensions/Asynchronous/ASYNCdialog.h

```diff
@@ -27,7 +27,9 @@ namespace enigma_user {
   int show_message_async(string str);
   int show_question_async(string str);
   int get_string_async(string message, string def, string cap="");
-  int get_integer_async(string message, string def, string cap="");
+  int get_password_async(string message, string def, string cap="");
+  int get_integer_async(string message, double def, string cap="");
+  int get_passcode_async(string message, double def, string cap="");
```

Comment:
@JoshDreamland To maintain cross-platform functionality, I'll have to change the name of this function in TFD as well, since you never complained about this before. In any case, what do you want me calling it? get_pin()?

Reply:
Well in Sam's defense, Apple calls it a "passcode"
https://support.apple.com/en-us/HT204060

Pin is also a nice alias for it too though that has meaning outside of computer devices. This isn't a big deal to me, I wouldn't care either way if I was using it myself, pin is shorter. I can't think of any distinction between pin and passcode that we could make that would justify us actually having both as a function. Perhaps offering pin by just making it an overload (make `get_pin` call `get_passcode` basically) is a useful alias.

Reply:
That's actually why I called it passcode, that's what my iPodTouch calls it. This is computer related, so why should we pretend it isn't?

Reply:
Well, do give me some slack for trying to interpret what Josh meant. But one reason is just maybe making it easier for novice programmers. Why calling it "pin" is easier for novices is because if they know what functionality they want and they are familiar with real life banking terminology (e.g, pin number of a credit card) then they will immediately understand what the function's purpose is when they see it.

But anyway, that's just one argument for calling it that. I really personally don't care and I don't think Josh does either, I think he was just suggesting it. Josh seems to like short function names too (pin is 3 characters, passcode is 8). But that only really matters for functions you expect people to use a lot repeatedly (like `d3d_model_vertex` or something).